src/test/java/ua/com/juja/microservices/teams/slackbot/aspect/DebugAroundExceptionLoggerTest.java

```diff
@@ -0,0 +1,155 @@
+import ua.com.juja.microservices.teams.slackbot.exceptions.UserExchangeException;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+import static org.powermock.api.mockito.PowerMockito.mock;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+
+/**
+ * @author Ivan Shapovalov
+ */
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({LoggerFactory.class})
+public class DebugAroundExceptionLoggerTest {
+    @Mock
+    private static Logger mockLogger;
+    private static Map<LogLine, Integer> warnLogLines = new HashMap<>();
+    private DebugAroundExceptionLogger debugAroundExceptionLogger = new DebugAroundExceptionLogger();
+    @Mock
+    private ProceedingJoinPoint proceedingJoinPoint;
+
+    @BeforeClass
+    public static void beforeClass() {
+
+        mockStatic(LoggerFactory.class);
+        mockLogger = mock(Logger.class);
+        when(LoggerFactory.getLogger(any(Class.class))).thenReturn(mockLogger);
+    }
+
+    @AfterClass
+    public static void checkLogsAfterClass() {
+        warnLogLines.forEach((key, value) -> verify(mockLogger, times(value))
+                .warn(key.getFormat(), key.getArg1(), key.getArg2()));
+    }
+
+    @Test
+    public void logExceptionHandleMethodIfOneArgumentWhichIsBaseBotExceptionWithErrorExecutedCorrectly() throws Throwable {
+
+        String exceptionMessage = "Some error";
+        ApiError apiError = new ApiError(
+                400, "USF-F1-D1",
+                exceptionMessage,
+                "User not found",
+                "Something went wrong",
+                Collections.emptyList()
+        );
+
+        BaseBotException exception = new UserExchangeException(apiError, new RuntimeException(exceptionMessage));
+        Object expected = new Object();
+        Object[] args = {exception};
+        String message = "execution(method)";
+        String warnMessage = "{} called with args '{}'!";
+
+        when(proceedingJoinPoint.getArgs()).thenReturn(args);
+        when(proceedingJoinPoint.toShortString()).thenReturn(message);
+        when(proceedingJoinPoint.proceed()).thenReturn(expected);
+
+        Object actual = debugAroundExceptionLogger.logExceptionHandleMethods(proceedingJoinPoint);
+
+        verify(proceedingJoinPoint).getArgs();
+        verify(proceedingJoinPoint).toShortString();
+        verify(proceedingJoinPoint).proceed();
+        addLogLineToLogs(warnMessage, message.replace("execution", "exception"),
+                exception.detailMessage());
+        verifyNoMoreInteractions(proceedingJoinPoint);
+        assertThat(actual, is(expected));
```

Comment:
In production call.proceed() return result of original wrapped method execution. In test it returns as we wish.